src/main/java/org/elasticsearch/cluster/metadata/MetaData.java

```diff
@@ -675,6 +675,16 @@ public String resolveIndexRouting(@Nullable String routing, String aliasOrIndex)
 
             aliasesOrIndices = convertFromWildcards(aliasesOrIndices, indicesOptions);
         }
+
+        if (aliasesOrIndices == null || aliasesOrIndices.length == 0) {
+            if (!indicesOptions.allowNoIndices()) {
+                throw new ElasticsearchIllegalArgumentException("Null or zero length argument for list of index or alias names "
```

Comment:
but now that I read your comment below, I see why it makes sense. Fine with illegal argument, I would only change the message to something like "no indices were specified but wildcard expansion is disabled" , shorter and more user oriented, what do you think?